### DISCUSSION TITLE
Function parameters are not ignored

### DISCUSSION THREAD

**Discussion by ehmjaysee** [peripheryapp/periphery#723] [Q&A]
Getting lots of warnings about unused function parameters like the one below. My shell command specifies --retain-unused-protocol-func-params so I was expecting these warnings to be suppressed. What am I missing?

warning: Parameter 'url' is unused

```
   func logFileSnapshot( url: URL )
    {
    }
```

My shell command:

/opt/homebrew/bin/periphery scan --retain-unused-protocol-func-params --workspace MotionIQ.xcworkspace --schemes MotionIQ  --targets MotionIQ  --format xcode 

Xcode 15.2
periphery 2.18.0

**Comment by ileitch** [comment]
@ehmjaysee Can you please create a small project that demonstrates the issue and attach it to a new issue? Thanks.

**Comment by ehmjaysee** [comment]
I see my mistake now. I was expecting all function parameters to be ignored, however the command will only ignore function parameters which are declared in protocols. Closing this issue now.
